### 1 задание
  Откройте набор данных. Посмотрите на содержимое, выведя в одержимое нескольких строчек

In [1]:
import pandas as pd

df = pd.read_csv('./mental_health_and_technology_usage_2024.csv')
df.head(5)

User_ID  Age  Gender  Technology_Usage_Hours  Social_Media_Usage_Hours  \
0  USER-00001   23  Female                    6.57                      6.00   
1  USER-00002   21    Male                    3.01                      2.57   
2  USER-00003   51    Male                    3.04                      6.14   
3  USER-00004   25  Female                    3.84                      4.48   
4  USER-00005   53    Male                    1.20                      0.56   

   Gaming_Hours  Screen_Time_Hours Mental_Health_Status Stress_Level  \
0          0.68              12.36                 Good          Low   
1          3.74               7.61                 Poor         High   
2          1.26               3.16                 Fair         High   
3          2.59              13.08            Excellent       Medium   
4          0.29              12.63                 Good          Low   

   Sleep_Hours  Physical_Activity_Hours Support_Systems_Access  \
0         8.01                     6.71                     No   
1         7.28                     5.88                    Yes   
2         8.04                     9.81                     No   
3         5.62                     5.28                    Yes   
4         5.55                     4.00                     No   

  Work_Environment_Impact Online_Support_Usage  
0                Negative                  Yes  
1                Positive                   No  
2                Negative                   No  
3                Negative                  Yes  
4                Positive                  Yes

### 2 задание
  Посмотрите с помощью методов Pandas на количество строчек и столбцов, информацию о столбцах (info)
  
  * User_ID: ID.
  * Age: Возраст участника.
  * Daily_Screen_Time (hours): Среднее ежедневное время экрана в часах.
  * Mental_Health_Score (1-10): Оценка психического здоровья по самооценке, где 1 – плохо, 10 – отлично.
  * Stress_Level (1-10): Уровень стресса по самооценке, где 1 — очень низкий, а 10 — очень высокий.
  * Sleep_Quality (1-10): Самооценка качества сна..

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   User_ID                   10000 non-null  object 
 1   Age                       10000 non-null  int64  
 2   Gender                    10000 non-null  object 
 3   Technology_Usage_Hours    10000 non-null  float64
 4   Social_Media_Usage_Hours  10000 non-null  float64
 5   Gaming_Hours              10000 non-null  float64
 6   Screen_Time_Hours         10000 non-null  float64
 7   Mental_Health_Status      10000 non-null  object 
 8   Stress_Level              10000 non-null  object 
 9   Sleep_Hours               10000 non-null  float64
 10  Physical_Activity_Hours   10000 non-null  float64
 11  Support_Systems_Access    10000 non-null  object 
 12  Work_Environment_Impact   10000 non-null  object 
 13  Online_Support_Usage      10000 non-null  object 
dtypes: floa

### 3 задание
  Выведите название столбцов и сопоставьте им русские названия в текстовой ячейке

In [3]:
df.columns.tolist()

['User_ID',
 'Age',
 'Gender',
 'Technology_Usage_Hours',
 'Social_Media_Usage_Hours',
 'Gaming_Hours',
 'Screen_Time_Hours',
 'Mental_Health_Status',
 'Stress_Level',
 'Sleep_Hours',
 'Physical_Activity_Hours',
 'Support_Systems_Access',
 'Work_Environment_Impact',
 'Online_Support_Usage']

1. User_ID — Идентификатор пользователя
2. Age — Возраст
3. Gender — Пол
4. Technology_Usage_Hours — Часы использования технологий
5. Social_Media_Usage_Hours — Часы использования социальных сетей
6. Gaming_Hours — Часы, потраченные на игры
7. Screen_Time_Hours — Время перед экраном (в часах)
8. Mental_Health_Status — Состояние психического здоровья
9. Stress_Level — Уровень стресса
10. Sleep_Hours — Часы сна
11. Physical_Activity_Hours — Часы физической активности
12. Support_Systems_Access — Доступ к системам поддержки
13. Work_Environment_Impact — Влияние рабочей среды
14. Online_Support_Usage — Использование онлайн-поддержки

### 4 задание
  Посмотрите описание (describe) числовых столбцов. Сделайте вывод  о наборе по минимальному и максимальному значениям, по среднему значению, по значению квартилей

In [4]:
df.describe()

Age  Technology_Usage_Hours  Social_Media_Usage_Hours  \
count  10000.000000            10000.000000              10000.000000   
mean      41.518600                6.474341                  3.972321   
std       13.920217                3.169022                  2.313707   
min       18.000000                1.000000                  0.000000   
25%       29.000000                3.760000                  1.980000   
50%       42.000000                6.425000                  3.950000   
75%       54.000000                9.212500                  5.990000   
max       65.000000               12.000000                  8.000000   

       Gaming_Hours  Screen_Time_Hours   Sleep_Hours  Physical_Activity_Hours  
count  10000.000000       10000.000000  10000.000000             10000.000000  
mean       2.515598           7.975765      6.500724                 5.003860  
std        1.446748           4.042608      1.450933                 2.905044  
min        0.000000           1.000000      4.000000                 0.000000  
25%        1.260000           4.520000      5.260000                 2.490000  
50%        2.520000           7.900000      6.500000                 4.990000  
75%        3.790000          11.500000      7.760000                 7.540000  
max        5.000000          15.000000      9.000000                10.000000

### Вывод по описанию числовых столбцов:

1. **Возраст (Age):**
   - Минимальный возраст: 18 лет
   - Максимальный возраст: 65 лет
   - Средний возраст: 41.5 года
   - 25-й перцентиль: 29 лет (четверть людей младше 29 лет)
   - 50-й перцентиль (медиана): 42 года (половина людей младше 42 лет)
   - 75-й перцентиль: 54 года (четверть людей старше 54 лет)

2. **Часы использования технологий (Technology_Usage_Hours):**
   - Минимум: 1 час
   - Максимум: 12 часов
   - Среднее значение: 6.47 часа в день
   - 25-й перцентиль: 3.76 часа
   - 50-й перцентиль: 6.43 часа
   - 75-й перцентиль: 9.21 часа

3. **Часы использования социальных сетей (Social_Media_Usage_Hours):**
   - Минимум: 0 часов
   - Максимум: 8 часов
   - Среднее значение: 3.97 часа
   - 25-й перцентиль: 1.98 часа
   - 50-й перцентиль: 3.95 часа
   - 75-й перцентиль: 5.99 часа

4. **Часы, потраченные на игры (Gaming_Hours):**
   - Минимум: 0 часов
   - Максимум: 5 часов
   - Среднее значение: 2.52 часа
   - 25-й перцентиль: 1.26 часа
   - 50-й перцентиль: 2.52 часа
   - 75-й перцентиль: 3.79 часа

5. **Время перед экраном (Screen_Time_Hours):**
   - Минимум: 1 час
   - Максимум: 15 часов
   - Среднее значение: 7.98 часа
   - 25-й перцентиль: 4.52 часа
   - 50-й перцентиль: 7.90 часа
   - 75-й перцентиль: 11.5 часа

6. **Часы сна (Sleep_Hours):**
   - Минимум: 4 часа
   - Максимум: 9 часов
   - Среднее значение: 6.5 часа
   - 25-й перцентиль: 5.26 часа
   - 50-й перцентиль: 6.5 часа
   - 75-й перцентиль: 7.76 часа

7. **Часы физической активности (Physical_Activity_Hours):**
   - Минимум: 0 часов
   - Максимум: 10 часов
   - Среднее значение: 5 часов
   - 25-й перцентиль: 2.49 часа
   - 50-й перцентиль: 4.99 часа
   - 75-й перцентиль: 7.54 часа

### Общий вывод:
- В выборке довольно широкий диапазон возрастов (от 18 до 65 лет) с медианой около 42 лет.
- Люди тратят в среднем около 6.5 часов в день на использование технологий, 4 часа — на социальные сети и около 2.5 часов — на игры.
- Среднее время перед экраном почти 8 часов, что довольно высоко.
- Среднее время сна составляет 6.5 часов, что ниже рекомендуемой нормы.
- Физическая активность в среднем составляет 5 часов, но у четверти выборки она составляет менее 2.5 часов в день.


### 5 задание
Посмотрите описание (describe) категориальных / текстовых столбцов. Сделайте выводы. 

In [5]:
df.describe(include=['object','category'])

User_ID Gender Mental_Health_Status Stress_Level  \
count        10000  10000                10000        10000   
unique       10000      3                    4            3   
top     USER-09984  Other            Excellent       Medium   
freq             1   3364                 2518         3338   

       Support_Systems_Access Work_Environment_Impact Online_Support_Usage  
count                   10000                   10000                10000  
unique                      2                       3                    2  
top                        No                Negative                   No  
freq                     5006                    3378                 5013

### Выводы:
- Наиболее часто встречающийся пол — "Other", возможно большинство воздержалось указывать).
- "Excellent" — самое частое состояние психического здоровья (25.18%).
- "Medium" — самый частый уровень стресса (33.38%).
- Около половины выборки (50%) не имеют доступа к системам поддержки.
- Около 33% респондентов оценивают влияние рабочей среды как негативное.
- Около половины людей не использует онлайн-поддержку.

### 6 задание
Для категориальных признаков посмотрите на уникальные значения этих признаков.

In [6]:
cols = df.describe(include=['object','category']).columns.tolist()
for col in cols[1:]:
    print(col, ': ', df[col].unique())

Gender :  ['Female' 'Male' 'Other']
Mental_Health_Status :  ['Good' 'Poor' 'Fair' 'Excellent']
Stress_Level :  ['Low' 'High' 'Medium']
Support_Systems_Access :  ['No' 'Yes']
Work_Environment_Impact :  ['Negative' 'Positive' 'Neutral']
Online_Support_Usage :  ['Yes' 'No']


### 7 задание
Попытайтесь выявить зависимость зависимости между психологическим состоянием ( Mental_Health_Status) и такими признаками как Gaming_Hours, Screen_Time_Hours, Sleep_Hours. Используйте группировку (по Mental_Health_Status - обязательно, по остальным признакам по желанию) и агрегацию данных

In [7]:
grouped_data_health = df.groupby('Mental_Health_Status')[['Gaming_Hours', 'Screen_Time_Hours', 'Sleep_Hours']].mean()
grouped_data_health

Gaming_Hours  Screen_Time_Hours  Sleep_Hours
Mental_Health_Status                                              
Excellent                 2.542133           7.949023     6.480175
Fair                      2.494956           7.920964     6.509289
Good                      2.503596           8.087707     6.497145
Poor                      2.521510           7.944783     6.516582

In [8]:
grouped_data_stress = df.groupby('Stress_Level')[['Gaming_Hours', 'Screen_Time_Hours', 'Sleep_Hours']].mean()
grouped_data_stress

Gaming_Hours  Screen_Time_Hours  Sleep_Hours
Stress_Level                                              
High              2.512459           7.918111     6.525204
Low               2.532011           8.078079     6.484361
Medium            2.502346           7.931150     6.492636

In [9]:
df.groupby('Mental_Health_Status').describe().T

Mental_Health_Status              Excellent         Fair         Good  \
Age                      count  2518.000000  2490.000000  2508.000000   
                         mean     41.551231    42.006827    41.237241   
                         std      13.863471    14.036469    13.875046   
                         min      18.000000    18.000000    18.000000   
                         25%      30.000000    29.000000    29.000000   
                         50%      42.000000    42.000000    41.000000   
                         75%      53.000000    54.000000    53.000000   
                         max      65.000000    65.000000    65.000000   
Technology_Usage_Hours   count  2518.000000  2490.000000  2508.000000   
                         mean      6.438995     6.352145     6.536304   
                         std       3.143397     3.209587     3.146783   
                         min       1.000000     1.000000     1.000000   
                         25%       3.810000     3.570000     3.840000   
                         50%       6.425000     6.180000     6.550000   
                         75%       9.130000     9.137500     9.252500   
                         max      11.990000    12.000000    12.000000   
Social_Media_Usage_Hours count  2518.000000  2490.000000  2508.000000   
                         mean      3.929365     4.023783     3.986523   
                         std       2.301193     2.304675     2.327314   
                         min       0.000000     0.000000     0.000000   
                         25%       1.980000     2.012500     1.940000   
                         50%       3.920000     4.035000     3.935000   
                         75%       5.950000     5.997500     6.020000   
                         max       8.000000     7.990000     8.000000   
Gaming_Hours             count  2518.000000  2490.000000  2508.000000   
                         mean      2.542133     2.494956     2.503596   
                         std       1.450221     1.438065     1.455652   
                         min       0.000000     0.000000     0.000000   
                         25%       1.302500     1.270000     1.210000   
                         50%       2.530000     2.510000     2.510000   
                         75%       3.850000     3.760000     3.800000   
                         max       5.000000     5.000000     5.000000   
Screen_Time_Hours        count  2518.000000  2490.000000  2508.000000   
                         mean      7.949023     7.920964     8.087707   
                         std       4.059197     4.003676     4.087353   
                         min       1.010000     1.000000     1.000000   
                         25%       4.410000     4.540000     4.590000   
                         50%       7.860000     7.940000     8.020000   
                         75%      11.545000    11.267500    11.742500   
                         max      15.000000    14.990000    15.000000   
Sleep_Hours              count  2518.000000  2490.000000  2508.000000   
                         mean      6.480175     6.509289     6.497145   
                         std       1.458015     1.449285     1.437176   
                         min       4.000000     4.000000     4.000000   
                         25%       5.200000     5.260000     5.307500   
                         50%       6.500000     6.540000     6.450000   
                         75%       7.750000     7.750000     7.730000   
                         max       9.000000     9.000000     9.000000   
Physical_Activity_Hours  count  2518.000000  2490.000000  2508.000000   
                         mean      5.054631     4.977277     4.964613   
                         std       2.890783     2.934293     2.918785   
                         min       0.000000     0.000000     0.010000   
                         25%       2.550000     2.380000     2.457500   
                         50%       5.070000     4.960000  

## Выводы

Как показывает таблицы, уровень стресса и уровень ментального здоровья практически не зависят от часов, проведенных за экраном и за играми, также не зависит и от часов сна.